tempodb/encoding/common/bloom.go

```diff
@@ -48,16 +81,20 @@ func (b *ShardedBloomFilter) WriteTo() ([][]byte, error) {
 	return bloomBytes, nil
 }
 
-func ShardKeyForTraceID(traceID []byte) int {
-	return int(util.TokenForTraceID(traceID)) % shardNum
-}
-
 // Test implements bloom.Test -> required only for testing
 func (b *ShardedBloomFilter) Test(traceID []byte) bool {
-	shardKey := ShardKeyForTraceID(traceID)
+	shardKey := ShardKeyForTraceID(traceID, len(b.blooms))
 	return b.blooms[shardKey].Test(traceID)
 }
 
-func GetShardNum() int {
-	return shardNum
+func ShardKeyForTraceID(traceID []byte, shardCount int) int {
+	return int(util.TokenForTraceID(traceID)) % ValidateShardCount(shardCount)
+}
+
+// For backward compatibility
+func ValidateShardCount(shardCount int) int {
```

Comment:
i think this functionality would be better placed in the marshalling code for BlockMeta. If the field does not exist then choose `legacyShardCount`.

Reply:
Agree on this but unfortunately, moving `ValidateShardCount` the block meta file results in a cyclic import.. 
```
package github.com/grafana/tempo/tempodb/encoding
	imports github.com/grafana/tempo/tempodb/backend
	imports github.com/grafana/tempo/tempodb/encoding/common
	imports github.com/grafana/tempo/tempodb/backend: import cycle not allowed
```
Will place it back here